### DISCUSSION TITLE
Pause and resume animation

### DISCUSSION THREAD

**Discussion by haophan1507** [software-mansion/react-native-reanimated#5013] [Q&A]
Hello, I'm using withRepeat to loop an animation. Now I want to pause and resume it. It seems the cancelAnimation function is not working as I want it to. Anyone have a solution?

**Comment by fukemy** [comment]
did you take a look here: https://docs.swmansion.com/react-native-reanimated/docs/core/cancelAnimation
 it's work as expected.

**Reply by haophan1507** [reply]
Thanks, I watched, but when restarting the motion, it starts at the stop, rather than starting from the beginning: 
https://github.com/software-mansion/react-native-reanimated/issues/4889

**Reply by LoserAntbear** [reply]
> did you take a look here: https://docs.swmansion.com/react-native-reanimated/docs/core/cancelAnimation it's work as expected.

The given example produces a bug, where repeated animation breaks the position after being paused.
If you pause translation animation halfway through and resume - it will be looped only to this position.
 

https://github.com/user-attachments/assets/ee7d080b-34f1-4749-a8cf-6151f971461f

**Comment by fukemy** [comment]
it's seem the only way is pause the `usesharedvalue` from change

**Reply by LoserAntbear** [reply]
Yea, something like that.
Or to write a custom hook.
